src/PackageCompiler.jl

```diff
@@ -700,14 +706,24 @@ function create_app(package_dir::String,
     package_name = ctx.env.pkg.name
     project = dirname(ctx.env.project_file)
 
+    # add precompile statements for functions that will be called from the C main() wrapper
+    precompiles = String[]
+    for (_, julia_main) in executables
+        push!(precompiles, "precompile(Tuple{typeof($package_name.$julia_main)})")
```

Comment:
CI errors because there is a check to see that a good error message is thrown in case one tries to create an executable from a function that doesn't exist. This could either be guarded with an `isdefined` or throw a good error here in case the function name doesn't exist and fixup the CI check correspondingly.